### PR TITLE
Remove dependency on xbean-finder

### PR DESCRIPTION
**Pull request by jkschneider**
Perform annotation classpath scanning with a vanilla ASM ClassVisitor.

Further work could be done in the future that reads class bytecodes directly, with a potential performance improvement.  See similar attempts at https://github.com/lukehutch/fast-classpath-scanner and https://github.com/rmuller/infomas-asl/tree/master/annotation-detector.  This would remove even the ASM dependency at the cost of having to update with future changes in bytecodes.

Removed mostly unused dependency on findbugs.  In particular, the findbugs-core dependency on findbugs-annotations:2.0.0 will break Java 8 consumers.

### CODE REVIEW

**Comment by elandau** [comment]
Any thoughts on using Guava's classpath scanner instead of asm?

File: governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    {
+        if(clazz == null)
+            clazz = classFromInternalName(className);
+        return clazz;
+    }
+
+    private Class<?> classFromInternalName(String name)
+    {
+        try {
+            return Class.forName(name.replace('/', '.'), true, classLoader);
```
Comment (elandau) [inline governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java]: Shouldn't the 2nd parameter be false so we don't force static initialization?

File: governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.objectweb.asm.Type.*;
+
+public class AnnotationFinder extends ClassVisitor
+{
+    private Set<Type> annotationTypes;
```
Comment (elandau) [inline governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java]: Make final

File: governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                                case BYTE: argClasses[i] = byte.class; break;
+                                case CHAR: argClasses[i] = char.class; break;
+                                case DOUBLE: argClasses[i] = double.class; break;
+                                case FLOAT: argClasses[i] = float.class; break;
+                                case INT: argClasses[i] = int.class; break;
+                                case LONG: argClasses[i] = long.class; break;
+                                case SHORT: argClasses[i] = short.class; break;
+                            }
+                        } catch(NullPointerException e) {
+                            System.out.println("here");
```
Comment (elandau) [inline governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java]: Log an error using slf4j if it's safe to ignore or throw an exception

File: governator-legacy/src/main/java/com/netflix/governator/lifecycle/ClasspathScanner.java [hunk 4]
```diff
@@ -114,38 +113,66 @@ public ClasspathScanner(Collection<String> basePackages, Collection<Class<? exte
...
+                                    localClasses.addAll(finder.getAnnotatedClasses());
+                                    localMethods.addAll(finder.getAnnotatedMethods());
+                                    localConstructors.addAll(finder.getAnnotatedConstructors());
+                                    localFields.addAll(finder.getAnnotatedFields());
+                                }
+                            }
+                        }
+                        catch( IOException e )
+                        {
+                            throw new IllegalStateException(e);
```
Comment (elandau) [inline governator-legacy/src/main/java/com/netflix/governator/lifecycle/ClasspathScanner.java]: Add jar name to exception to help debug errors

File: governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java (hunk 1 above)
Comment (jkschneider) [inline governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java]: In thought that as well, but was following what xbean-finder was doing internally.

File: governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java (hunk 3 above)
Comment (jkschneider) [inline governator-legacy/src/main/java/com/netflix/governator/lifecycle/AnnotationFinder.java]: Sorry, this was a temporary debugging issue, will delete.

**Comment by jkschneider** [comment]
Concerning Guava's classpath scanning utility, their docs say this:

> It's worth noting that ClassPath is a best-effort utility. It only scans classes in jar files or under a file system directory. Neither can it scan classes managed by custom class loaders that aren't URLClassLoader. **So don't use it for mission critical production tasks.**

File: governator-legacy/src/main/java/com/netflix/governator/lifecycle/ClasspathScanner.java (hunk 4 above)
Comment (jkschneider) [inline governator-legacy/src/main/java/com/netflix/governator/lifecycle/ClasspathScanner.java]: Thanks.  This also brought to my attention that I wasn't closing the JarFile resource.  Oops!

**Pull request merged**